### PR TITLE
feat: reprovider warning and announce-on|off profiles

### PR DESCRIPTION
**Pull request by lidel**
### TLDR

This PR aims to fix footgun that is `lowpower` profile + warn users who already sabotaged themselves by disabling reprovide system (not announcing to DHT).

### Details

Users running on raspberrypi or other "lowpower" server often sabotaged themselves by disabling announcements when applying poorly documented `lowpower` profile ([example](https://discuss.ipfs.tech/t/understanding-long-access-times/18514/3)). 

To make matters worse,  we've been prominently hinting that user should enable this profile to save resources, we did that in examples and docs ([example](http://web.archive.org/web/20240924162820/https://docs.ipfs.tech/install/command-line/#system-requirements) – i've hidden this footgun from our docs, but the harm is already done)

This PR moves announcement settings to a separate profile, removing the footgun from `lowpower` one + warns users who have it disabled (if it was intentional, user can ignore the message).

### TODO

- [x] announce-on/off profiles
- [x] `lowprofile` should not impact announcements
- [x] `ipfs daemon` should print warning when node is online but reprovide system is turned off
- [x] update docs/config
- [x] add changelog entry

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -299,7 +299,7 @@ Map of HTTP headers to set on responses from the RPC (`/api/v0`) HTTP server.
 Example:
 ```json
 {
-	"Foo": ["bar"]
+  "Foo": ["bar"]
```
Comment (lidel) [inline docs/config.md:302]: :information_source:  replaced tabs with space in entire file for consistency

File: docs/changelogs/v0.31.md [hunk 2]
```diff
@@ -6,13 +6,26 @@
...
 
+#### `lowpower` profile no longer breaks DHT announcements
+
+We've notices users were applying `lowpower` profile, and then reporting content routing issues. This was because `lowpower` disabled reprovider system and locally hosted data was no longer announced on Amino DHT.
+
+This release changes [`lowpower` profile](https://github.com/ipfs/kubo/blob/master/docs/config.md#lowpower-profile) to not change reprovider settings, ensuring the new users are not sabotaging themselves. It also adds [`annouce-on`](https://github.com/ipfs/kubo/blob/master/docs/config.md#announce-on-profile) and [`announce-off`](https://github.com/ipfs/kubo/blob/master/docs/config.md#announce-off-profile) profiles for controlling announcement settings separately.
+
+> [!IMPORTANT]
+> If you've ever applied the `lowpower` profile before, there is a high chance your node is not announcing to DHT anymore.
+> If you have `Reprovider.Interval` set to `0` you may want to wet it to `22h` (or run `ipfs config profile apply announce-on`) to fix your system.
```
Comment (jaller94) [inline docs/changelogs/v0.31.md:25]: Typo: wet -> set
